resources/js/Components/LanguageSelector.vue

```diff
@@ -73,22 +74,22 @@ const shownLanguages = computed<Language[]>(() => {
 
 function onInput(searchedLanguage: string): void {
 	searchInput.value = searchedLanguage;
-    getApiLanguageCodes(searchInput.value);
+    debouncedApiLanguageSearch(searchInput.value);
 	highlightedIndex.value = 0;
 }
 
-async function getApiLanguageCodes(inputValue: string) {
-    return await axios.get(
+const debouncedApiLanguageSearch = debounce(async (debouncedInputValue: string) => {
+    await axios.get(
         'https://www.wikidata.org/w/api.php?action=languagesearch&format=json&formatversion=2',
         {
             params: {
-                search: inputValue,
+                search: debouncedInputValue,
                 origin: '*' // avoid CORS console errors
             }
         }).then((response) => {
             apiLanguageCodes.value = Object.keys(response.data.languagesearch);
         });
-}
+}, 200);
```

Comment:
For me, 200ms is good, but should we give a heads-up to the UX team?

Reply:
I discussed this last week with Sarai and it was decided there is not need to plan changes to the UX flow until deemed necessary in product verification. see [here](https://phabricator.wikimedia.org/T353727#9554935).